### PR TITLE
feat(vault): support None type

### PR DESCRIPTION
**Pull request by tranhuyducseven**

### CODE REVIEW

File: ui/utils/rule.ts [hunk 1]
```diff
@@ -66,6 +66,21 @@ export const isNumericType = (type: TRuleType): boolean => {
   }
 };
 
+export const isBytesType = (type: TRuleType): boolean => {
+  switch (type) {
+    case 'bytes1':
+    case 'bytes2':
```
Comment (vinhtrand8) [inline ui/utils/rule.ts:72]: add case`bytes3`

File: ui/utils/rule.ts [hunk 2]
```diff
@@ -108,102 +123,200 @@ export const getMaxValue = (type: TRuleType): number => {
...
-  );
+export const isNBytesValue = (type: TRuleType, val: string): boolean => {
+  let num = 0;
+  switch (type) {
+    case 'bytes1':
+      num = 1;
+      break;
+    case 'bytes2':
+      num = 2;
+      break;
```
Comment (vinhtrand8) [inline ui/utils/rule.ts:134]: add case `bytes3`

File: ui/utils/rule.ts [hunk 3]
```diff
@@ -108,102 +123,200 @@ export const getMaxValue = (type: TRuleType): number => {
...
+    .transform((val) => val.trim())
+    .optional(),
+  bytes1: z
+    .string()
+    .transform((val) => val.trim())
+    .optional(),
+  bytes2: z
+    .string()
+    .transform((val) => val.trim())
+    .optional(),
```
Comment (vinhtrand8) [inline ui/utils/rule.ts:265]: add `bytes3`

**Comment by david7art** [comment]
ok

**Review by david7art — APPROVED** [review]

**Review by vinhtrand8 — APPROVED** [review]

**Pull request merged**